lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -662,51 +669,32 @@ def unblock_command(cmd)
   # This code was originally taken from https://github.com/ruby/ruby/blob/93420d34aaf8c30f11a66dd08eb186da922c831d/lib/shellwords.rb#L88
   #
   def shellsplitex(line)
-    quote = nil
-    words = []
+    unbalanced_quote = nil
+    tokens = []
     field = String.new
-    regexp = %r{
-      \G\s*(?>(?<word>[^\s\'\"]+)             # Words within str
+    current_field_begin = nil
 
-      |                                       # OR
+    line.scan(/\G(\s*)(?>([^\s\\\'\"]+)|'([^\']*)'|"((?:[^\"\\]|\\.)*)"|(\\.?)|(\S))(\s|\z)?/m) do |preceding_whitespace, word, sq, dq, esc, garbage, sep|
+      next if unbalanced_quote.present?
 
-      '(?<sq>[^\']*)'                         # Text between single quotes
-
-      |                                       # OR
-
-      "(?<dq>(?:[^\"\\]|\\.)*)"               # Text between double quotes
-
-      |                                       # OR
-
-      (?<esc>\\.?)                            # Escapes used on special characters
-
-      |                                       # OR
-
-      (?<garbage>\S))                         # Anything that wasn't already matched, expect whitespace
-
-      (?<sep>\s|\z)?                          # Separators
-    }ix
-
-    line.scan(regexp) do
-      |word, sq, dq, esc, garbage, sep|
+      current_field_begin ||= Regexp.last_match.begin(0) + preceding_whitespace.length
       if garbage
-        if quote.nil?
-          quote = garbage
-        else
-          field << garbage
-        end
-        next
+        quote_start_begin = Regexp.last_match.begin(0)
+        unbalanced_quote = garbage
+        field << line[quote_start_begin + 1..-1]
+        break
```

Comment:
Now if there's garbage: Instead of continuing to parse the remaining string, we just return a single token with the remaining value